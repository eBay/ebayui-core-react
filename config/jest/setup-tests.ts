import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
