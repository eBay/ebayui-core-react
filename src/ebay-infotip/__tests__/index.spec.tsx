import React from 'react'
import { render, fireEvent, RenderResult } from '@testing-library/react'
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayInfotip, EbayInfotipContent, EbayInfotipHeading } from '../../index'

jest.mock('../../common/random-id', () => ({ randomId: () => 'abc123' }))

const renderComponent = (props?: any) => render(
    <EbayInfotip {...props}>
        <EbayInfotipHeading>Title</EbayInfotipHeading>
        <EbayInfotipContent>
            <p>Info content</p>
        </EbayInfotipContent>
    </EbayInfotip>
)

const checkIsExpanded = (wrapper: RenderResult) => {
    expect(wrapper.container.querySelector('.infotip--expanded')).toBeInTheDocument()
    expect(wrapper.container.querySelector('button[aria-expanded=true]')).toBeInTheDocument()
}

const checkIsCollapsed = (wrapper: RenderResult) => {
    expect(wrapper.container.querySelector('.infotip--expanded')).not.toBeInTheDocument()
    expect(wrapper.container.querySelector('button[aria-expanded=true]')).not.toBeInTheDocument()
}

describe('<EbayInfotip>', () => {
    describe('on infotip button click', () => {
        it('should expand the infotip', () => {
            const wrapper = renderComponent()
            fireEvent.click(wrapper.container.querySelector('button.infotip__host'))
            checkIsExpanded(wrapper)
        })
    })

    describe('on infotip button close click', () => {
        it('should collapse the infotip', () => {
            const wrapper = renderComponent()
            fireEvent.click(wrapper.container.querySelector('.infotip__close'))
            checkIsCollapsed(wrapper)
        })
    })

    describe('on infotip expanded', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const wrapper = renderComponent({ onExpand: spy })
            fireEvent.click(wrapper.container.querySelector('button.infotip__host'))

            expect(spy).toBeCalled()
        })
    })

    describe('on infotip collapsed', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const wrapper = renderComponent({ onCollapse: spy })
            fireEvent.click(wrapper.container.querySelector('button.infotip__host'))
            fireEvent.click(wrapper.container.querySelector('button.infotip__close'))

            expect(spy).toBeCalled()
        })
    })

    describe('on using the infotip with no content', () => {
        it('should throw an error', () => {
            jest.spyOn(console, 'error').mockImplementation(() => null)
            expect(() => {
                render(<EbayInfotip a11yCloseText="Close" />)
            }).toThrowError(`EbayInfotip: Please use a EbayInfotipContent that defines the content of the infotip`);
            (console.error as jest.Mock).mockRestore()
        })
    })

    describe('on using the close button', () => {
        it('should pass the property to the button that close the infotip', () => {
            const wrapper = render(
                <EbayInfotip a11yCloseText="Dismiss info">
                    <EbayInfotipHeading>Title</EbayInfotipHeading>
                    <EbayInfotipContent>
                        <p>Info content</p>
                    </EbayInfotipContent>
                </EbayInfotip>
            )

            expect(wrapper.container.querySelector('.infotip__close[aria-label="Dismiss info"]')).toBeInTheDocument()
        })

        it('should focus the button element', () => {
            const { getByLabelText } = render(
                <EbayInfotip a11yCloseText="Dismiss info" aria-label="info">
                    <EbayInfotipContent>
                        <p>Info content</p>
                    </EbayInfotipContent>
                </EbayInfotip>
            )

            const host = getByLabelText('info')

            fireEvent.click(host)
            expect(host).not.toHaveFocus()

            const close = getByLabelText('Dismiss info')

            fireEvent.click(close)
            expect(host).toHaveFocus()
        })
    })
})

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
})
