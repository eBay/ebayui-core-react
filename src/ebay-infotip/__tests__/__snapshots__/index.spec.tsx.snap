// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-infotip Custom button content (With render prop) 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 200px;"
  >
    <span
      class="infotip"
    >
      <button
        aria-expanded="false"
        aria-label="Click to open infotip"
        class="icon-btn icon-btn--transparent infotip__host"
        style="height: auto; width: auto;"
        type="button"
      >
        <span
          style="display: inline-flex; align-items: center;"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
          <span
            style="margin-left: 5px;"
          >
            Click me
          </span>
        </span>
      </button>
      <span
        class="infotip__overlay"
        style="transform: scale3d(1,1,1); left: -10px;"
      >
        <span
          class="infotip__pointer infotip__pointer--top-left"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Use Access Key 'S' to display settings.
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Custom icon 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 200px;"
  >
    <span
      class="infotip"
    >
      <button
        aria-expanded="false"
        aria-label="Infotip"
        class="icon-btn icon-btn--transparent infotip__host"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--settings-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-settings-16"
          />
        </svg>
      </button>
      <span
        class="infotip__overlay"
        style="transform: translateX(-50%) scale3d(1,1,1); left: 50%;"
      >
        <span
          class="infotip__pointer infotip__pointer--bottom"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Content
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Default 1`] = `
<DocumentFragment>
  <div
    style="display: flex; margin: 200px;"
  >
    <span
      class="infotip"
    >
      <button
        aria-expanded="false"
        aria-label="Infotip"
        class="icon-btn icon-btn--transparent infotip__host"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--information-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-information-16"
          />
        </svg>
      </button>
      <span
        class="infotip__overlay"
        style="transform: translateX(-50%) scale3d(1,1,1); left: 50%;"
      >
        <span
          class="infotip__pointer infotip__pointer--bottom"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Content
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Disabled 1`] = `
<DocumentFragment>
  <div
    style="display: flex; margin: 200px;"
  >
    <span
      class="infotip"
    >
      <button
        aria-expanded="false"
        aria-label="Infotip"
        class="icon-btn icon-btn--transparent infotip__host"
        disabled=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--information-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-information-16"
          />
        </svg>
      </button>
      <span
        class="infotip__overlay"
        style="transform: translateX(-50%) scale3d(1,1,1); left: 50%;"
      >
        <span
          class="infotip__pointer infotip__pointer--bottom"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Content
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Expanded by default 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 200px;"
  >
    <span
      class="infotip infotip--expanded"
    >
      <button
        aria-expanded="true"
        aria-label="Infotip"
        class="icon-btn icon-btn--transparent infotip__host"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--information-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-information-16"
          />
        </svg>
      </button>
      <span
        class="infotip__overlay"
        style="transform: scale3d(1,1,1); left: -10px;"
      >
        <span
          class="infotip__pointer infotip__pointer--top-left"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Use Access Key 'S' to display settings.
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip In paragraph 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 100px;"
  >
    <em>
      NOTE: No block elements can be nested in p elements, like div, h1-6, or other p elements. Any content with that will break
    </em>
    <p>
      Some paragraph content 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Important information"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(-50%) scale3d(1,1,1); left: 50%;"
        >
          <span
            class="infotip__pointer infotip__pointer--bottom"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Important
                </span>
                <span>
                  This is some important info
                </span>
              </span>
              <button
                aria-label="Dismiss infotip"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
       More paragraph content
    </p>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Modal 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 100px;"
  >
    <span
      class="dialog--mini infotip"
    >
      <button
        aria-expanded="false"
        aria-label="Infotip"
        class="icon-btn icon-btn--transparent infotip__host dialog--mini__host"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--information-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-information-16"
          />
        </svg>
      </button>
    </span>
    <div>
      <div
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="drawer-dialog dialog--mini__overlay drawer-dialog--mask-fade-slow"
        hidden=""
        mode="mini"
        role="dialog"
      >
        <div
          class="drawer-dialog__window drawer-dialog__window drawer-dialog__window--slide"
        >
          <button
            aria-label="Maximize Text Label"
            class="drawer-dialog__handle"
            type="button"
          />
          <div
            class="drawer-dialog__header"
          >
            <h2
              id="dialog-title-abc123"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
            </h2>
            <button
              aria-label="Close"
              class="icon-btn drawer-dialog__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </div>
          <div
            class="drawer-dialog__main"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Pointer direction 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin-left: 300px;"
  >
    <div
      style="margin: 100px 0px;"
    >
      top 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(-50%) scale3d(1,1,1); left: 50%;"
        >
          <span
            class="infotip__pointer infotip__pointer--top"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      top-left 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: scale3d(1,1,1); left: -10px;"
        >
          <span
            class="infotip__pointer infotip__pointer--top-left"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      top-right 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: scale3d(1,1,1); right: -10px;"
        >
          <span
            class="infotip__pointer infotip__pointer--top-right"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      right 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(-16px) translateY(-50%) scale3d(1,1,1); right: 100%; top: -6px;"
        >
          <span
            class="infotip__pointer infotip__pointer--right"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      right-bottom 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(-16px) scale3d(1,1,1); right: 100%; bottom: -50%;"
        >
          <span
            class="infotip__pointer infotip__pointer--right-bottom"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      right-top 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(-16px) scale3d(1,1,1); right: 100%; top: -100%;"
        >
          <span
            class="infotip__pointer infotip__pointer--right-top"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      bottom 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(-50%) scale3d(1,1,1); left: 50%;"
        >
          <span
            class="infotip__pointer infotip__pointer--bottom"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      bottom-left 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: scale3d(1,1,1); left: -10px;"
        >
          <span
            class="infotip__pointer infotip__pointer--bottom-left"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      bottom-right 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: scale3d(1,1,1); right: -10px;"
        >
          <span
            class="infotip__pointer infotip__pointer--bottom-right"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      left 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(16px) translateY(-50%) scale3d(1,1,1); left: 100%; top: -6px;"
        >
          <span
            class="infotip__pointer infotip__pointer--left"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      left-bottom 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(16px) scale3d(1,1,1); left: 100%; bottom: -10px;"
        >
          <span
            class="infotip__pointer infotip__pointer--left-bottom"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      style="margin: 100px 0px;"
    >
      left-top 
      <span
        class="infotip"
      >
        <button
          aria-expanded="false"
          aria-label="Infotip"
          class="icon-btn icon-btn--transparent infotip__host"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="icon icon--information-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-information-16"
            />
          </svg>
        </button>
        <span
          class="infotip__overlay"
          style="transform: translateX(16px) scale3d(1,1,1); left: 100%; top: -100%;"
        >
          <span
            class="infotip__pointer infotip__pointer--left-top"
          />
          <span
            class="infotip__mask"
          >
            <span
              class="infotip__cell"
            >
              <span
                class="infotip__content"
              >
                <span
                  class="infotip__heading"
                >
                  Title
                </span>
                <p>
                  Use Access Key 'S' to display settings.
                </p>
              </span>
              <button
                aria-label="Close"
                class="icon-btn icon-btn--transparent infotip__close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="icon icon--close-16"
                  focusable="false"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    xlink:href="#icon-close-16"
                  />
                </svg>
              </button>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Pointer with custom location 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 200px;"
  >
    <span
      class="infotip"
    >
      <button
        aria-expanded="false"
        aria-label="Infotip"
        class="icon-btn icon-btn--transparent infotip__host"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--information-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-information-16"
          />
        </svg>
      </button>
      <span
        class="infotip__overlay"
        style="transform: scale3d(1,1,1); left: -16px; top: 40px;"
      >
        <span
          class="infotip__pointer infotip__pointer--top-left"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Use Access Key 'S' to display settings.
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-infotip Text instead of icon 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; margin: 200px;"
  >
    <span
      class="infotip"
    >
      <button
        aria-expanded="false"
        class="icon-btn icon-btn--transparent infotip__host"
        type="button"
      >
        Click for infotip
      </button>
      <span
        class="infotip__overlay"
        style="transform: scale3d(1,1,1); left: -10px;"
      >
        <span
          class="infotip__pointer infotip__pointer--top-left"
        />
        <span
          class="infotip__mask"
        >
          <span
            class="infotip__cell"
          >
            <span
              class="infotip__content"
            >
              <span
                class="infotip__heading"
              >
                Title
              </span>
              <p>
                Use Access Key 'S' to display settings.
              </p>
            </span>
            <button
              aria-label="Close"
              class="icon-btn icon-btn--transparent infotip__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close-16"
                focusable="false"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close-16"
                />
              </svg>
            </button>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
